### PR TITLE
Implement CurveZMQ security for the network

### PR DESCRIPTION
**Pull request by jsmitchell**
This implements an authentication handshake and transport encryption using ZMQ's CurveZMQ support (ec based on edwards). The support is implemented in the shared Interconnect object, but is only enabled for the network (validator to validator) instance. In the future, we can also enable this for the intercomponent instance, but that will require changes to 'clients'.

### CODE REVIEW

File: validator/sawtooth_validator/networking/interconnect.py [hunk 1]
```diff
@@ -39,11 +42,16 @@ def get_enum_name(enum_value):
 
 
 class _SendReceive(object):
-    def __init__(self, address, futures, identity=None, dispatcher=None):
+    def __init__(self, address, futures, identity=None,
+                 dispatcher=None, secured=False,
+                 server_public_key=None, server_private_key=None):
```
Comment (peterschwarz) [inline validator/sawtooth_validator/networking/interconnect.py]: Can we document these parameters?  Are they strings? File names?

File: validator/sawtooth_validator/networking/interconnect.py [hunk 2]
```diff
@@ -143,20 +181,30 @@ def __init__(self,
                  endpoint,
                  dispatcher,
                  identity=None,
-                 peer_connections=None):
+                 peer_connections=None,
+                 secured=False,
+                 server_public_key=None,
+                 server_private_key=None):
```
Comment (peterschwarz) [inline validator/sawtooth_validator/networking/interconnect.py]: Doc here on these fields

File: validator/sawtooth_validator/networking/interconnect.py (hunk 1 above)
Comment (jsmitchell) [inline validator/sawtooth_validator/networking/interconnect.py]: Added in squashed commit 3448bb7

File: validator/sawtooth_validator/networking/interconnect.py (hunk 2 above)
Comment (jsmitchell) [inline validator/sawtooth_validator/networking/interconnect.py]: Added in squashed commit 3448bb7

**Pull request merged**